### DISCUSSION TITLE
Match batch total with file total

### DISCUSSION THREAD

**Discussion by palsri2004** [moov-io/ach#1360] [General]
Hello,

I need to process a nacha file and for that need to validate if all totals are matching with file total.Is there a function in moov library to check if all batch totals are matching file total or would that be a custom logic? Thank you.

**Comment by adamdecaf** [comment]
The library computes batch/file totals when you call `.Create()`, so you can use that to get the total amounts.

**Reply by palsri2004** [reply]
Thank you! So I already have an ACH file in memory and in order to validate totals of that file I need to call create method?

**Reply by adamdecaf** [reply]
Yea, see https://pkg.go.dev/github.com/moov-io/ach#File.Create

> Create will modify the File to tabulate and assemble it into a valid state. This includes setting any posting dates, sequence numbers, counts, and sums.
